ir/src/constraints/list_folding.rs

```diff
@@ -0,0 +1,67 @@
+use super::{
+    graph::CURRENT_ROW, list_comprehension::unfold_lc, IdentifierType, SemanticError, SymbolTable,
+};
+use air_script_core::{
+    ConstantType, Expression, IndexedTraceAccess, ListFoldingValueType, VariableType,
+};
```

Comment:
nit: I'd prefer to import these from super so that we can see everything that is used from `air_script_core` by looking at the `lib.rs` file at the top of the crate